### PR TITLE
pastebinit: change upstream to maintained fork, add tests, and manpage

### PR DESCRIPTION
**Pull request by Samuel-Martineau**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/pastebinit/default.nix [hunk 1]
```diff
@@ -3,47 +3,74 @@
...
+      fi
+    done
   '';
 
   meta = with lib; {
-    homepage = "https://launchpad.net/pastebinit";
+    homepage = "https://launchpad.net/ubuntu/+source/pastebinit";
     description = "A software that lets you send anything you want directly to a pastebin from the command line";
-    maintainers = with maintainers; [ raboof ];
+    maintainers = with maintainers; [ raboof samuel-martineau ];
```
Comment (raboof) [inline pkgs/tools/misc/pastebinit/default.nix:73]: Thanks for adding yourself as maintainer!

File: pkgs/tools/misc/pastebinit/default.nix [hunk 2]
```diff
@@ -3,47 +3,74 @@
...
+, docbook-xsl-ns
+, installShellFiles
+, wget
+, cacert
 }:
-stdenv.mkDerivation rec {
-  version = "1.5";
+stdenv.mkDerivation
+rec {
+  version = "1.6.2-1";
```
Comment (raboof) [inline pkgs/tools/misc/pastebinit/default.nix:14]: I think the `-1` postfix is part of the ubuntu/debian packaging and doesn't need to be in the Nix package

File: pkgs/tools/misc/pastebinit/default.nix [hunk 3]
```diff
@@ -3,47 +3,74 @@
...
     cp -a pastebinit $out/bin
     cp -a pastebin.d $out/etc
-    substituteInPlace $out/bin/pastebinit --replace "'/etc/pastebin.d" "'$out/etc/pastebin.d"
+    substituteInPlace $out/bin/pastebinit --subst-var-by "etc" "$out/etc"
+    installManPage pastebinit.1
+  '';
+
+  doCheck = true;
+  nativeCheckInputs = buildInputs ++ [ wget cacert ];
+  checkPhase = ''
```
Comment (raboof) [inline pkgs/tools/misc/pastebinit/default.nix:50]: This check does not require any 'intermediate' build results, but only the 'installed' package, right? In that case I think it belongs in `passthru.tests` instead (see for example https://nixos.org/manual/nixpkgs/unstable/#var-meta-tests, but we should definitely improve those docs)

File: pkgs/tools/misc/pastebinit/default.nix [hunk 4]
```diff
@@ -3,47 +3,74 @@
...
+    export USER="nixpkgs-pastebinit-tests"
+    teststring="test from pastebinit"
+    for pastebin in $(ls pastebin.d/*.conf | sed 's~^pastebin.d/\(.*\)\.conf$~\1~'); do
+      echo "Trying $pastebin"
+      URL=$(printf "$teststring\n$teststring\n$teststring" | python ./pastebinit -b $pastebin)
```
Comment (raboof) [inline pkgs/tools/misc/pastebinit/default.nix:55]: A bit weird to publish stuff to the internet as part of the tests, which might be another reason to move it to `passthru.tests`

File: pkgs/tools/misc/pastebinit/default.nix (hunk 3 above)
Comment (Samuel-Martineau) [reply pkgs/tools/misc/pastebinit/default.nix]: Done!

File: pkgs/tools/misc/pastebinit/default.nix (hunk 4 above)
Comment (Samuel-Martineau) [reply pkgs/tools/misc/pastebinit/default.nix]: I took inspiration from the upstream tests

**Review by raboof — APPROVED** [review]
LGTM, but needs a conflict fix

**Comment by Samuel-Martineau** [comment]
@raboof I made a mistake whilst trying to fix the merge conflict and accidentally ended up deleting the branch of my fork. I didn't seem to have an option to reopen this pull request now that the branch exists again, so I opened #270974 which contains the same changes, but can merge correctly.

**Pull request closed without merging**